h5i_identifier.go

```diff
@@ -44,13 +44,13 @@ func (i Identifier) Name() string {
 		return ""
 	}
 	buf := string(make([]byte, sz))
-	c_buf := C.CString(buf)
-	defer C.free(unsafe.Pointer(c_buf))
-	sz = int(C.H5Iget_name(i.id, c_buf, C.size_t(sz)))
+	cBuf := C.CString(buf)
```

Comment:
Why is this not just `C.malloc(sz+1)`?

Reply:
Hmm. Then we have to start casting `sz + 1` to `C.size_t` and `cBuf` to `*C.char`. The question is what is more beautiful?